testing/trino-tests/src/test/java/io/trino/tests/TestExcludeColumnsFunction.java

```diff
@@ -0,0 +1,221 @@
+    {
+        assertThatThrownBy(() -> query("""
+                SELECT *
+                FROM TABLE(exclude_columns(
+                                    input => TABLE(tpch.tiny.nation),
+                                    columns => CAST(null AS DESCRIPTOR)))
+                """))
+                .hasMessage("Excluded columns descriptor is null");
+
+        assertThatThrownBy(() -> query("""
+                SELECT *
+                FROM TABLE(exclude_columns(
+                                    input => TABLE(tpch.tiny.nation),
+                                    columns => DESCRIPTOR()))
+                """))
+                .hasMessage("line 4:21: Invalid descriptor argument COLUMNS. Descriptors should be formatted as 'DESCRIPTOR(name [type], ...)'");
+
+        assertThatThrownBy(() -> query("""
+                SELECT *
+                FROM TABLE(exclude_columns(
+                                    input => TABLE(tpch.tiny.nation),
+                                    columns => DESCRIPTOR(foo, comment, bar)))
+                """))
+                .hasMessage("Excluded columns: [foo, bar] not present in the table");
+
+        assertThatThrownBy(() -> query("""
+                SELECT *
+                FROM TABLE(exclude_columns(
+                                    input => TABLE(tpch.tiny.nation),
+                                    columns => DESCRIPTOR(nationkey bigint, comment)))
+                """))
+                .hasMessage("Excluded columns descriptor contains types");
+
+        assertThatThrownBy(() -> query("""
+                SELECT *
+                FROM TABLE(exclude_columns(
+                                    input => TABLE(tpch.tiny.nation),
+                                    columns => DESCRIPTOR(nationkey, name, regionkey, comment)))
+                """))
+                .hasMessage("All columns are excluded");
+    }
+
+    @Test
+    public void testColumnResolution()
+    {
+        // excluded column names are matched case-insensitive
+        assertThat(query("""
+                SELECT *
+                FROM TABLE(exclude_columns(
+                                    input => TABLE(SELECT 1, 2, 3, 4, 5) t(a, B, "c", "D", e),
+                                    columns => DESCRIPTOR("A", "b", C, d)))
+                """))
+                .matches("SELECT 5");
+    }
+
+    @Test
+    public void testReturnedColumnNames()
+    {
+        // the function preserves the incoming column names. However, due to how the analyzer handles identifiers, these are not the canonical names according to the SQL identifier semantics.
+        assertThat(query("""
+                SELECT *
+                FROM TABLE(exclude_columns(
+                                    input => TABLE(SELECT 1, 2, 3, 4, 5) t(a, B, "c", "D", e),
+                                    columns => DESCRIPTOR(e)))
+                """))
+                .hasColumnNames("a", "B", "c", "D");
```

Comment:
What if the subquery contains an anonymous column?

Reply:
You cannot exclude an anonymous column. Such column is returned as anonymous output column.

Reply:
I know. My point is that this kind of test wouldn’t work with anonymous columns involved, as hasColumnNames cannot test for that. See my comment above. TLDR, I don’t think it’s the right way to test for column names.